aw_core/transforms.py

```diff
@@ -43,14 +43,16 @@ def heartbeat_reduce(events: Event, pulsetime: float) -> List[Event]:
 
 
 def heartbeat_merge(last_event: Event, heartbeat: Event, pulsetime: float) -> Optional[Event]:
-    """Merges two events together if they have identical labels and are separated by a time smaller than pulsetime."""
+    """
+    Merges two events if they have identical labels and are
+    separated by a time smaller than :code:`pulsetime` seconds.
+    """
     if last_event.data == heartbeat.data:
-        last_event_end = last_event.timestamp + last_event.duration
-        pulsewindow_end = last_event_end + timedelta(seconds=pulsetime)
+        gap = heartbeat.timestamp - (last_event.timestamp + last_event.duration)
```

Comment:
It would be easier if the duration would be included when calculating the gap so the API user doesn't have to keep track of increasing the pulsetime all the time. In that case we could even skip the pulsetime altogether and only rely on the duration. What would be confusing in that case though would be that a heartbeat event ends with timestamp and a normal event starts with the timestamp. Both solutions have their flaws, just wanted to mention it.

Reply:
The programmer shouldn't have to increase the pulsetime as the duration of the heartbeat grows. Or perhaps I misunderstood?

Reply:
It was I who misunderstood how heartbeats with duration worked.